account-kit/core/src/store/store.ts

```diff
@@ -76,10 +79,10 @@ export const createAccountKitStore = (
                   if (!persistedConnection) return false;
 
                   return Array.from(Object.entries(c)).every(([key, value]) => {
-                    return (
-                      (persistedConnection as Record<string, any>)[key] ===
-                      value
-                    );
+                    const pConn = persistedConnection as Record<string, any>;
+                    if (key === "chain") return value.id === pConn.chain.id;
+
+                    return pConn[key] === value;
```

Comment:
yaayyy testing, we need to get better at our coverage here. it's really bad and this core logic is does a lot of complex state mgmt that needs to be tested.